apps/prairielearn/src/lib/client/cookie.ts

```diff
@@ -0,0 +1,45 @@
+import CookiesModule from 'js-cookie';
+
+// Keep in sync with assets/scripts/navbarClient.ts
+const COOKIE_EXPIRATION_DAYS = 30;
+
+let OldCookies: Cookies.CookiesStatic<string> | undefined;
+let Cookies: Cookies.CookiesStatic<string> | undefined;
+
+// Allows for server-side rendering.
+if (typeof location !== 'undefined') {
+  OldCookies = CookiesModule.withAttributes({
+    path: '/',
+    expires: COOKIE_EXPIRATION_DAYS,
+    secure: location.protocol === 'https:',
+  });
+
+  // New cookies do have a domain.
+  Cookies = CookiesModule.withAttributes({
+    path: '/',
+    expires: COOKIE_EXPIRATION_DAYS,
+    domain:
+      document.querySelector('meta[name="cookie-domain"]')?.getAttribute('content') ?? undefined,
+    secure: location.protocol === 'https:',
+  });
+}
+
+type OldAndNewCookieNames = [string, string];
+
+export function setCookieClient(names: OldAndNewCookieNames, value: string) {
+  if (!OldCookies || !Cookies) {
+    throw new Error('location is undefined');
```

Comment:
Nit, could "Cannot be used outside a browser environment" be a better error message?